### DISCUSSION TITLE
No anti-aliasing

### DISCUSSION THREAD

**Discussion by Shumpert820** [nothings/stb#1268] [Q&A] [answered]
hello,
i am doing a project using stb lib. The font need contour like this:
![image](https://user-images.githubusercontent.com/16200028/147620696-62c6371e-2447-4f0f-8d92-dd7f70c0223d.png)

```
    /* init font size */
    pixels = ft->font_size * 1.0;

    /* scale = pixels / (ascent - descent) */
    scale = stbtt_ScaleForPixelHeight(&stb_ctx.info, pixels);

    /* get vertical metrics */
    stbtt_GetFontVMetrics(&stb_ctx.info, &ascent, &descent, &lineGap);

    if (stb_ctx.image_buf != NULL)
    {
        stbtt_FreeSDF(stb_ctx.image_buf, NULL);
        stb_ctx.image_buf= NULL;
    }
   /* PPS_MEDIA_FONT_PADDING_SIZE 1 PPS_MEDIA_ALPHA_BLENDING_BG 90 */
    stb_ctx.image_buf = stbtt_GetCodepointSDF(&stb_ctx.info, scale, glyph, PPS_MEDIA_FONT_PADDING_SIZE, PPS_MEDIA_ALPHA_BLENDING_BG,
                              PPS_MEDIA_ALPHA_BLENDING_BG / PPS_MEDIA_FONT_PADDING_SIZE, &width, &height, &xoff, &yoff);

    /* get horizontal metrics */
    stbtt_GetCodepointHMetrics(&stb_ctx.info, (int)glyph, &advanceWidth, &leftSideBearing);
```
After I got image_buf, I transfer the buf val into argb1555 and put them into device canvas. did I miss any operations?
the first font size is 64 and the second is 16.
![image](https://user-images.githubusercontent.com/16200028/147621019-10f0620b-fd4e-42c5-a32d-cb65a5a6821c.png)
![image](https://user-images.githubusercontent.com/16200028/147620968-66f9a85f-0cae-41be-9702-ed6fbe8561bd.png)
transfer code:
```
    pixel_color.red   = value;
    pixel_color.green = value;
    pixel_color.blue  = value;
```

**Comment by nothings** [comment]
do you want a character bitmap or do you want a character SDF?

**Reply by Shumpert820** [reply]
Actually i want a character bitmap with dark edge.

**Reply by Shumpert820** [reply]
I looked into sdf_test.c which include antialias. However the formula render the whole character with dark color. How can I modify this formula to get the effect of the image below
![image](https://user-images.githubusercontent.com/16200028/147633539-7e852523-73de-4b90-9a2f-6ef2ac042024.png)

**Comment by nothings** [comment, marked as answer]
ok, almost none of your question makes sense.

first, stbtt is a tool, not a solution to your problem. you're going to have to do work to get to your solution, and I can't debug that for you over the internet. both stbtt bitmaps and SDFs are single-channel output from 0..255. the bitmaps are anti-aliased but don't have outlines, and the SDFs are SDFs so they require you to compute the antialiasing and outlines yourself. Surely there are dozens of blog posts on the internet about how to make text with outlines from SDFs? That's the reason I included SDF output in stbtt with no real docs, for people who already know about the idea but wanted SDF data to work with.

Second, nothing you've said explains how you're using stbtt. saying you're using a 1555-bit ARGB doesn't tell me where it's coming from or how you're getting it from the 8-bit single-channel output.

Third, I can tell you that you're not going to get good aliasing with a 1-bit alpha channel, regardless.
